### PR TITLE
[CI] Use quay as the default image registry

### PR DESCRIPTION
**Pull request by kevin85421**
## Why are these changes needed?

* Use Quay as the default image registry. See #1924 for more context.
* `manifests/` has a lot of overlap with `ray-operator/config/`, so I remove the directory.

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: docs/deploy/installation.md [hunk 1]
```diff
@@ -25,26 +22,6 @@ helm install kuberay-operator kuberay/kuberay-operator
...
-kubectl create -k "github.com/ray-project/kuberay/manifests/cluster-scope-resources?ref=v1.0.0&timeout=90s"
-
-# Install KubeRay operator
-kubectl apply -k "github.com/ray-project/kuberay/manifests/base?ref=v1.0.0"
-```
-
-> Observe that we must use `kubectl create` to install cluster-scoped resources.
-> The corresponding `kubectl apply` command will not work. See [KubeRay issue #271](https://github.com/ray-project/kuberay/issues/271).
-
-### Single Namespace version
```
Comment (kevin85421) [inline docs/deploy/installation.md:38]: If users want to use this feature, users should use [Helm chart](https://docs.ray.io/en/latest/cluster/kubernetes/user-guides/helm-chart-rbac.html#kuberay-helm-chart-rbac) instead.

File: docs/deploy/installation.md [hunk 2]
```diff
@@ -25,26 +22,6 @@ helm install kuberay-operator kuberay/kuberay-operator
...
-```sh
-export KUBERAY_NAMESPACE=<my-awesome-namespace>
-
-# Install CRDs (Executed by cluster admin)
-kustomize build "github.com/ray-project/kuberay/manifests/overlays/single-namespace-resources?ref=v1.0.0" | envsubst | kubectl create -f -
-
-# Install KubeRay operator (Executed by user)
-kustomize build "github.com/ray-project/kuberay/manifests/overlays/single-namespace?ref=v1.0.0" | envsubst | kubectl apply -f -
+# Install CRD and KubeRay operator
+kubectl create -k "github.com/ray-project/kuberay/ray-operator/config/default?ref=v1.0.0&timeout=90s"
```
Comment (kevin85421) [inline docs/deploy/installation.md:26]: Screen Shot 2024-02-22 at 3 28 53 PM

File: docs/deploy/installation.md [hunk 3]
```diff
@@ -7,15 +7,12 @@ Make sure your Kubernetes cluster and Kubectl are both at version at least 1.23.
 ```sh
 export KUBERAY_VERSION=master
 
-# Install CRDs
-kubectl create -k "github.com/ray-project/kuberay/manifests/cluster-scope-resources?ref=${KUBERAY_VERSION}&timeout=90s"
-
-# Install KubeRay operator
-kubectl apply -k "github.com/ray-project/kuberay/manifests/base?ref=${KUBERAY_VERSION}&timeout=90s"
+# Install CRD and KubeRay operator
+kubectl create -k "github.com/ray-project/kuberay/ray-operator/config/default?ref=${KUBERAY_VERSION}&timeout=90s"
```
Comment (kevin85421) [inline docs/deploy/installation.md:11]: Screen Shot 2024-02-22 at 3 52 11 PM

File: .buildkite/test-sample-yamls.yml [hunk 4]
```diff
@@ -18,7 +18,7 @@
   commands:
     - ./.buildkite/setup-env.sh
     # Use KubeRay operator image from the latest release
-    - source .venv/bin/activate && BUILDKITE_ENV=true OPERATOR_IMAGE=kuberay/operator:v1.0.0 python3 tests/test_sample_raycluster_yamls.py
+    - source .venv/bin/activate && BUILDKITE_ENV=true OPERATOR_IMAGE=quay.io/kuberay/operator:v1.0.0 python3 tests/test_sample_raycluster_yamls.py
```
Comment (kevin85421) [inline .buildkite/test-sample-yamls.yml:21]: I run the following commands locally.

```sh
OPERATOR_IMAGE=quay.io/kuberay/operator:v1.0.0 python3 tests/test_sample_raycluster_yamls.py
# Check the KubeRay operator's image => quay.io/kuberay/operator:v1.0.0
```

File: helm-chart/kuberay-operator/README.md [hunk 5]
```diff
@@ -40,7 +40,7 @@ helm version
   * Use Helm's built-in `--skip-crds` flag to install the operator only. See [this document](https://helm.sh/docs/chart_best_practices/custom_resource_definitions/) for more details.
   ```sh
   # Step 1: Install CRDs only (for cluster admin)
-  kubectl create -k "github.com/ray-project/kuberay/manifests/cluster-scope-resources?ref=v1.0.0&timeout=90s"
+  kubectl create -k "github.com/ray-project/kuberay/ray-operator/config/crd?ref=v1.0.0&timeout=90s"
```
Comment (kevin85421) [inline helm-chart/kuberay-operator/README.md:43]: Screen Shot 2024-02-22 at 3 58 49 PM

File: helm-chart/kuberay-operator/values.yaml [hunk 6]
```diff
@@ -3,7 +3,7 @@
 # Declare variables to be passed into your templates.
 
 image:
-  repository: kuberay/operator
+  repository: quay.io/kuberay/operator
```
Comment (kevin85421) [inline helm-chart/kuberay-operator/values.yaml:6]: Install the KubeRay operator via Helm chart locally, and verify the image.

```sh
helm install kuberay-operator .
# Check the KubeRay operator's image => quay.io/kuberay/operator:nightly
```

File: ray-operator/config/default/kustomization.yaml [hunk 7]
```diff
@@ -22,6 +22,6 @@ bases:
 
 images:
 - name: kuberay/operator
-  newName: kuberay/operator
```
Comment (kevin85421) [inline ray-operator/config/default/kustomization.yaml:25]: Screen Shot 2024-02-22 at 4 40 17 PM

**Comment by kevin85421** [comment]
cc @Yicheng-Lu-llll would you mind reviewing this PR? Thanks!

**Review by aslonnie — COMMENTED** [review]
(I do not really have enough context for this, and I do not really care). I think you folks can decide?

**Review by Yicheng-Lu-llll — APPROVED** [review]
lgtm

**Comment by kevin85421** [comment]
> (I do not really have enough context for this, and I do not really care). I think you folks can decide?

Thanks! The community has already decided to use Quay as the default image registry for KubeRay v1.1.0 because (1) it doesn't have pull limit (2) it has more usage log.

**Review by jjyao — APPROVED** [review]

**Pull request merged**